src/adapter/src/coord/sequencer.rs

```diff
@@ -2315,6 +2269,51 @@ impl<S: Append + 'static> Coordinator<S> {
                 .timestamp
         };
 
+        if in_immediate_multi_stmt_txn {
+            if let Some(txn_reads) = self.txn_reads.get(&session.conn_id()) {
+                // Verify that the references and indexes for this query are in the
+                // current read transaction.
+                let allowed_id_bundle = txn_reads.read_holds.id_bundle();
+                if !allowed_id_bundle.is_empty() || !txn_reads.timestamp_independent {
```

Comment:
See https://github.com/MaterializeInc/materialize/pull/16470#discussion_r1039728321

Reply:
If the id bundle is empty shouldn't this produce an error, because the ids from the current query haven't had their compaction held back and will panic when queried at the txn's timestamp?

Reply:
This is starting to get convoluted/complicated and I think we should maybe think about simplifying peeks. However, the idea here is that IF the bundle is empty and the reads were timestamp independent, THEN the timestamp will be re-determined and a new read hold will be taken. This happens when the previous query is something constant like `SELECT 1`.

Reply:
When the query is `SELECT 1` we end up taking an empty read hold and then throwing it away later. We probably shouldn't be doing this in the first place.

Reply:
Ah right. I forgot about the timestamp independent stuff. Agree this is getting quite complicated.